### PR TITLE
SmtpMailer: constructor parameter $options changed to public properties

### PR DESCRIPTION
**Pull request by dg**

### CODE REVIEW

File: src/Mail/SmtpMailer.php [hunk 1]
```diff
@@ -51,10 +51,10 @@ public function __construct(array $options = [])
 		}
 		$this->username = isset($options['username']) ? $options['username'] : '';
 		$this->password = isset($options['password']) ? $options['password'] : '';
-		$this->secure = isset($options['secure']) ? $options['secure'] : '';
+		$this->encryption = isset($options['secure']) ? $options['secure'] : '';
 		$this->timeout = isset($options['timeout']) ? (int) $options['timeout'] : 20;
 		if (!$this->port) {
-			$this->port = $this->secure === 'ssl' ? 465 : 25;
+			$this->port = $this->encryption === 'ssl' ? 465 : 25;
```
Comment (fprochazka) [inline src/Mail/SmtpMailer.php]: maybe few constants would be nice?

**Comment by fprochazka** [comment]
Nice :+1:

**Comment by hrach** [comment]
Well, what is the benefit? I see it pretty problematic, changin the property doesn't close the old connection.

**Comment by fprochazka** [comment]
@hrach you'll get better error if you pass incorrect config options to the smtp, even without using the DIC extension

**Comment by f3l1x** [comment]
Why we can't use setters for this? :smiley: I'm not big friend of public properties at all, there's also username/password.

**Pull request closed without merging**